### DISCUSSION TITLE
Override DHCP on a secondary bridged interface?

### DISCUSSION THREAD

**Discussion by mbentley** [lima-vm/lima#1688] [Q&A] [answered]
I have a VM running using lima and I've added a secondary `bridged` network interface which works well except I am not sure how I would override the cidata config that is set for the network interface in a way that was either intended or supported.  Right now, I am just using a DHCP reservation but I also have other use cases for network interfaces (manually just setting a network interface up without other configuration - useful for me when using macvlan with containers to allow the host to communicate to the container.  For my last use case there, I _can_ just have the network interface come up and get an assigned DHCP address to achieve what I am looking for but I don't necessarily need that interface getting an IP.

I saw someone trying to assign an IP in https://github.com/lima-vm/lima/discussions/677 but I am not trying to change the primary interface.  I'm going to try something similar in the meantime but I am not 100% sure what happens when there are two definitions for the same network interface.

**Comment by BitWise-0x** [comment, marked as answer]
doesn't /etc/netplan/ *.yml 's work for this? It's what I've always used...

**Reply by mbentley** [reply]
Ah, I think I see why what I was trying to do isn't working.  I just want to bring the interface up without setting an IP.  It's taking my empty config:

```
network:
  version: 2
  renderer: networkd
  ethernets:
    lima1: {}
```

Seems like it's preferring a config file with any config over one with an empty config so just setting dhcp4: false seems to do what I want.

```
network:
  version: 2
  renderer: networkd
  ethernets:
    lima1:
      dhcp4: false
```

I thought I was hitting an issue with the order to precedence for the files but it seems to prefer some config over nothing.
